stor/tests/test_dx.py

```diff
@@ -210,47 +232,59 @@ def test_iterating_over_files(self):
             obj.write(b'hello')
+            obj.flush()
             obj.write(b' world')
-        self.assertIn(b'hello world', dx_p.read_object())
-
-    @mock.patch('time.sleep', autospec=True)
-    @mock.patch.object(DXPath, 'upload', autospec=True)
-    def test_write_multiple_flush_multiple_upload(self, mock_upload):
-        dx_p = DXPath('dx://project/obj')
-        with NamedTemporaryFile(delete=False) as ntf1,\
-                NamedTemporaryFile(delete=False) as ntf2,\
-                NamedTemporaryFile(delete=False) as ntf3:
-            with mock.patch('tempfile.NamedTemporaryFile', autospec=True) as ntf:
-                ntf.side_effect = [ntf1, ntf2, ntf3]
-                with dx_p.open(mode='wb') as obj:
-                    obj.write(b'hello')
-                    obj.flush()
-                    obj.write(b' world')
-                    obj.flush()
-                u1, u2, u3 = mock_upload.call_args_list
-                u1[0][1][0].source == ntf1.name
-                u2[0][1][0].source == ntf2.name
-                u3[0][1][0].source == ntf3.name
-                u1[0][1][0].object_name == dx_p.resource
-                u2[0][1][0].object_name == dx_p.resource
-                u3[0][1][0].object_name == dx_p.resource
-                self.assertEqual(open(ntf1.name).read(), 'hello')
-                self.assertEqual(open(ntf2.name).read(), 'hello world')
-                # third call happens because we don't care about checking for
-                # additional file change
-                self.assertEqual(open(ntf3.name).read(), 'hello world')
-
-
-@unittest.skip("skipping")
+            obj.flush()
+        f = dxpy.DXFile(dxid=dx_p.canonical_resource,
+                        project=dx_p.canonical_project)
+        f.wait_on_close(20)  # to allow for max of 20s for file state to go to closed
+        self.assertEqual(dx_p.open().read(), 'hello world')
+
+    def test_read_dir_fail(self):
+        self.setup_temporary_project()
+        self.setup_file('/temp_folder/file')
+        dx_p = DXPath('dx://' + self.project + ':/temp_folder')
+        with pytest.raises(dx.NotFoundError, match='No data object'):
+            dx_p.open().read()
+        dx_p = DXPath('dx://' + self.project)
+        with pytest.raises(ValueError, match='Cannot read project'):
+            dx_p.open().read()
+
+    def test_write_to_project_fail(self):
+        self.setup_temporary_project()
+        dx_p = DXPath('dx://' + self.project)
+        with pytest.raises(ValueError, match='Cannot write to project'):
+            with dx_p.open(mode='wb') as f:
+                f.write(b'data')
+
+
 class TestDXShared(SharedOBSFileCases, DXTestCase):
-    drive = 'dx://'
+    drive = 'dx://project:'  # project is required for DX paths
     path_class = DXPath
     normal_path = DXPath('dx://project:/obj')
 
+    def test_makedirs_p_does_nothing(self):
+        # skipping dumb test...because our project doesn't exist on DNAnexus
```

Comment:
I'm a little confused - I would think the reason you'd skip this is because makedirs_p *does* do something on DNAnexus.

Additionally, is this an integration test? I'm a little confused by naming here - are all `DXTestCase` auto-wrapped in vcrpy? If so, perhaps we could call them `DXRecordedTestCase` or something like that?

Reply:
`test_makedirs_p_does_nothing` is one of the tests in `SharedOBSFileCases` which are all run as part of this `TestDXShared` class. As you said, makedirs_p does do something on DX, as well as the fact that you need a project etc to run this test, hence I had to override the superClass test and essentially skip it.